src/server/mod.rs

```diff
@@ -726,12 +694,16 @@ impl LanguageServer for LspServer {
             Err(err) => return Err(err.into()),
         };
 
-        let node = find_node(
-            walk::Node::Package(&pkg),
+        let mut visitor = semantic::NodeFinderVisitor::new(
             params.text_document_position.position,
         );
+        flux::semantic::walk::walk(
+            &mut visitor,
+            walk::Node::Package(&pkg),
+        );
```

Comment:
Assume this decorates/mutates the AST with semantic information?

Reply:
If you look at the source for `find_node`, what we've replaced here is the contents of that function. No actual change in functionality. In this case, we're using a visitor to find a specific node.